### PR TITLE
add mac control layer #976

### LINKED ISSUE #976: Question: is it possible to generate Ethernet Pause frames
Hi, 

Is it possible to generate ethernet pause frames using this tool. 

in that case, can you please share some sample code for the same.

Regards,
Saravanankoori J

### PR DESCRIPTION
**Pull request by speakinghedge**
adds MACControl messages:
* Pause
* Gate
* Report
* Register/REQ/ACK
* Class Based Flow Control

based on the Wireshark dissector and [this](http://www.studioreti.it/slide/08_SwFlowContr_E_A.pdf) old slides.

fixes #976

### CODE REVIEW

File: scapy/contrib/mac_control.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+        return pkt + pay
+
+
+class MACControlPause(MACControl):
+    fields_desc = [
+        ShortField("_op_code", MACControl.OP_CODE_PAUSE),
+        ShortField("pause_time", 0),
+    ]
+
+    def get_pause_time_10_mbit(self):
```
Comment (guedou) [inline scapy/contrib/mac_control.py]: Looks redundant. Can't you use a single method that accept the data rate as an argument?

Comment (speakinghedge) [reply scapy/contrib/mac_control.py]: If functional decomposition goes mad... hope aa4ff56 fixes this issue in a reasonable way.

**Comment by speakinghedge** [comment]
did. seems everything is fine - test failed cause of some mumbo jumbo in L2ListenTcpdump.

**Comment by guedou** [comment]
Thanks for this PR. The issue with failed tests is temporary fixed with #1320

**Pull request merged**